Shared/Samples/Edit geodatabase with transactions/EditGeodatabaseWithTransactionsView.swift

```diff
@@ -0,0 +1,266 @@
+            }
+            .task(id: selectedAction) {
+                guard let selectedAction else { return }
+                
+                do {
+                    switch selectedAction {
+                    case .setUp:
+                        try await model.setUp()
+                    case .addFeature(let tableName, let featureTypeID):
+                        try await model.addFeature(
+                            to: tableName,
+                            featureTypeID: featureTypeID,
+                            point: tapLocation!
+                        )
+                    case .beginTransaction:
+                        try model.geodatabase.beginTransaction()
+                        isInTransaction = true
+                    case .commitTransaction:
+                        try model.geodatabase.commitTransaction()
+                        isInTransaction = false
+                    case .rollbackTransaction:
+                        try model.geodatabase.rollbackTransaction()
+                        isInTransaction = false
+                    case .sync:
+                        try await model.syncGeodatabase()
+                    }
+                    
+                    statusText = selectedAction.completionMessage
+                } catch {
+                    self.error = error
+                }
+                
+                self.selectedAction = nil
+            }
+            .overlay(alignment: .top) {
+                Text(statusText)
+                    .multilineTextAlignment(.center)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .overlay(alignment: .center) {
+                if let progress = model.progress {
+                    VStack {
+                        Text(selectedAction == .setUp ? "Creating geodatabase…" : "Syncing edits…")
+                            .padding(.bottom)
+                        ProgressView(progress)
+                            .frame(maxWidth: 180)
+                    }
+                    .padding()
+                    .background(.ultraThickMaterial)
+                    .cornerRadius(10)
+                    .shadow(radius: 50)
+                }
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Button(isInTransaction ? "End" : "Start") {
+                        if isInTransaction {
+                            // Presents the alert to handle the edits.
+                            endTransactionAlertIsPresented = true
+                        } else {
+                            selectedAction = .beginTransaction
+                        }
+                    }
+                    .disabled(!transactionIsRequired)
+                    .popover(isPresented: $isSelectingFeatureType) {
```

Comment:
ℹ️ This is fine if the `SelectFeatureTypeView` is presented by the "Start" button tap, but is weird if the "Requires Transaction" toggle is off and the view is triggered by a tap on the map view - because the source view for the popover is disabled.

Ideally the popover can be shown at the tapped screenpoint, but that logic is irrelevant to this sample, so I'm OK with keeping it as is.